1-Draft/RFCXXXX-Umask-and-Ulimit.md

```diff
@@ -0,0 +1,276 @@
+  DataSegmentSize 9223372036854775807 9223372036854775807
+        StackSize             8388608            67104768
+     CoreFileSize                   0 9223372036854775807
+  AddresSpaceSize 9223372036854775807 9223372036854775807
+MemoryAddressSize 9223372036854775807 9223372036854775807
+NumberOfProcesses                2837                4256
+NumberOfOpenFiles               10240 9223372036854775807
+
+
+PS> get-ulimit -all |get-member
+
+   TypeName: System.Management.Automation.Platform+ResourceLimitInfo
+Name        MemberType Definition
+----        ---------- ----------
+Equals      Method     bool Equals(System.Object obj)
+GetHashCode Method     int GetHashCode()
+GetType     Method     type GetType()
+ToString    Method     string ToString()
+Current     Property   ulong Current {get;set;}
+Maximum     Property   ulong Maximum {get;set;}
+Resource    Property   System.Management.Automation.Platform+ResourceForUlimit Resource {get;set;}
+```
+
+#### Implementation Notes
+
+The `ulimit` cmdlets do not totally replicate the Unix behavior.
+Specifically, resources which are not alterable via the `setrlimit` API will not be supported.
+The pipe size resource is not retrievable or alterable with the ulimit cmdlets.
+The resources available will vary based on the platform, the following table shows what resources shall be returned based on their platform:
+
+| Resource | MacOS | Linux |
+| -------- | :---: | :---: |
+| Maximum CPU Time | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum File Size | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum Data Segment Size | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum Stack Size | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum Core File Size | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum Resident Set Size | :heavy_check_mark: | :heavy_check_mark: |
+| Memory Address Space | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum number of processes | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum number of open Files | :heavy_check_mark: | :heavy_check_mark: |
+| Maximum locked-in-memory address space | :heavy_check_mark: | :heavy_check_mark: |
+| Address space limit | | :heavy_check_mark: |
+| Maximum number of file locks | | :heavy_check_mark: |
+| Maximum number of pending signals | | :heavy_check_mark: |
+| Maximum bytes in message queue | | :heavy_check_mark: |
+| Maximum nice priority allowed | | :heavy_check_mark: |
+| Maximum real-time priority | | :heavy_check_mark: |
+| Timeout for real-time tasks (microseconds) | | :heavy_check_mark: |
+
+#### Set-Ulimit Syntax
+
+```powershell
+NAME
+    Set-Ulimit
+
+SYNTAX
+    Set-Ulimit [-ResourceInfo] <Platform+ResourceLimitInfo> [<CommonParameters>]
+
+    Set-Ulimit -Resource {CpuTime | FileSize | DataSegmentSize | StackSize | CoreFileSize | AddresSpaceSize | MemoryAddressSize | 
+    NumberOfProcesses | NumberOfOpenFiles} [-Current <ulong>] [-Maximum <ulong>] [<CommonParameters>]
```

Comment:
I think it's worth explaining how `-Current` relates to  `-Maximum`.

Presumably, `-Current` is what https://ss64.com/bash/ulimit.html calls the _soft_ limit, and `-Maximum` corresponds to the _hard_ limit.

Generally, the help will have to explain this mapping.


Note that Bash's `ulimit` by default sets _both_ limits.
If we want to deviate from that by only setting the _soft_ (`-Current`) limit by default, we should make that clear (I think it makes more sense, but it is a departure).

Reply:
Also,

* adding a property to reflect the _units_ of the resource values would be helpful; from https://ss64.com/bash/ulimit.html:

> values are in 1024-byte increments, except for '-t', which is in seconds, '-p', which is in units of 512-byte blocks, and '-n' and '-u', which are unscaled values.

* consider supporting the symbolic `soft`, `hard` and `unlimited`  values that Bash's `ulimit` supports.

Reply:
`Soft` and `Hard` might be worthy aliases for `Current` and `Maximum`